modules/drivers/mongo/test/metabase/driver/mongo/parameters_test.clj

```diff
@@ -480,3 +480,98 @@
                                                                             :snippet-name "first 3 checkins"
                                                                             :snippet-id   (:id snippet)}}}
                   :parameters []}))))))))
+
+(deftest ^:parallel field-filter-variable-with-alias-test
+  (testing "native query with field filter variable and a field alias works"
+    (mt/test-driver :mongo
+      (let [query (mt/native-query
+                   {:collection "orders"
+                    :query (json/encode
+                            [{"$lookup" {"from" "products",
+                                         "let" {"let_product_id___1" "$product_id"},
+                                         "pipeline" [{"$project" {"_id" "$_id",
+                                                                  "category" "$category"}},
+                                                     {"$match" {"$expr" {"$eq" ["$$let_product_id___1",
+                                                                                "$_id"]}}}],
+                                         "as" "join_alias_Products"}},
+                             {"$unwind" {"path" "$join_alias_Products",
+                                         "preserveNullAndEmptyArrays" true}},
+                             {"$match" (json/raw-json-generator "{{category_filter}}")},
+                             {"$group" {"_id" nil,
+                                        "count" {"$sum" 1}}},
+                             {"$sort" {"_id" 1}},
+                             {"$project" {"_id" false,
+                                          "count" true}}])
+                    :template-tags {"category_filter" {:id "category_filter_id"
+                                                       :name "category_filter"
+                                                       :display-name "Category Filter"
+                                                       :type :dimension
+                                                       :dimension [:field (mt/id :products :category) nil]
+                                                       :alias "join_alias_Products.category"
+                                                       :widget-type :string/=
+                                                       :default ["Gizmo"]}}})]
+        (is (= [[4784]]
```

Comment:
This asserts too much. The original bug was that it returned no results. The complex query plus exact rows returned makes it hard to understand. I would change it to `(is (seq (-> ...`